torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1234,7 +1234,7 @@ void check_gpu_tensors(const std::vector<at::Tensor>& tensors) {
       TORCH_CHECK(false, "Tensors must have identical strides");
     }
     if (!t.is_non_overlapping_and_dense()) {
```

Comment:
do you have an idea about what to change this check to in the end-state, or how to make it supported for lazy?

Reply:
LazyTensor should be `non_overlapping_and_dense`. But there is a bug preventing us from doing that: https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/core/tensor_impl.cpp#L78.